packages/alea-frontend/components/CoverageUpdater.tsx

```diff
@@ -184,7 +206,40 @@ export function CoverageUpdater({ courseId, snaps, secInfo, handleSave }: Covera
         </Box>
       )}
       <Dialog open={editDialogOpen} onClose={handleEditDialogClose} maxWidth="md" fullWidth>
-        <DialogTitle>Edit Coverage Entry</DialogTitle>
+        <DialogTitle
+          sx={{ display: 'flex', alignItems: 'center', justifyContent: 'space-between' }}
+        >
+          Edit Coverage Entry
+          {formData.autoDetected && (
+            <Box sx={{ display: 'flex', alignItems: 'center', gap: 1 }}>
+              <Typography variant="body2" color="warning.main" fontWeight="bold">
+                Auto-detected Data
+              </Typography>
+              <Tooltip
```

Comment:
This seems very similar to the component on the Coverage Table. Please refactor out a component and try to reduce copy-paste as much as possible.